### PR TITLE
fix or suppress test warnings

### PR DESCRIPTION
**Pull request by mathause**
Fixes or suppresses a number of warnings that turn up in our upstream CI.

`pd.Index.is_monotonic` is an alias for `pd.Index.is_monotonic_increasing` and does _not_ stand for `pd.Index.is_monotonic_increasing or pd.Index.is_monotonic_decreasing`.

### CODE REVIEW

**Comment by max-sixty** [comment]
Thanks a lot!

File: xarray/core/groupby.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def _unique_and_monotonic(group):
     if isinstance(group, _DummyGroup):
         return True
     index = safe_cast_to_index(group)
-    return index.is_unique and index.is_monotonic
+    return index.is_unique and index.is_monotonic_increasing
```
Comment (mathause) [inline xarray/core/groupby.py:204]: @dcherian do you know if `is_monotonic_increasing` correct here or should that be `(is_monotonic_increasing or is_monotonic_decreasing)`.

(Note that `pd.Index.is_monotonic` is an alias for `pd.Index.is_monotonic_increasing`).

Comment (dcherian) [reply xarray/core/groupby.py:204]: `(is_monotonic_increasing or is_monotonic_decreasing)` I think this is the intent (see lines 355-362 below where we avoid calling pd.factorize when the indices are unique and monotonic) but I doubt it's tested. I think we just end up using the "slow" path with more copies for monotonically decreasing group variables.

Comment (dcherian) [reply xarray/core/groupby.py:204]: Opened #6220

Comment (mathause) [reply xarray/core/groupby.py:204]: Thanks! Let's merge this as is then.

**Pull request merged**